src/gen_bpf.c

```diff
@@ -2254,6 +2251,9 @@ struct bpf_program *gen_bpf_generate(const struct db_filter_col *col)
 	struct bpf_state state;
 	struct bpf_program *prgm;
 
+	if (col->filter_cnt == 0)
+		return NULL;
```

Comment:
Returning NULL here will lead to us returning `-ENOMEM`.  The comment in `_rc_filter()` says this about `-ENOMEM`:

```
/* NOTE: unable to allocate enough memory to perform the
*       requested operation */
```
I am afraid this will cause confusion for some users.  Should we update the above comment?  Or is it fair to return `-EINVAL` because the user tried to build a bpf program with no valid filters in it?

Reply:
This is tricky as currently `gen_bpf_generate(...)` doesn't really offer a lot of granularity when it comes to return value: it's either a valid pointer or NULL.  If there is any hope, it is that `_gen_bpf_build_bpf(...)` does return an error code and not simply a pointer; we could modify `gen_bpf_generate(...)` to return the `bpf_program` pointer via an argument and return a more useful return code ... assuming we got back and audit all the return codes in "src/gen_bpf.c".

What do you think @drakenclimber?

Reply:
I just added another patch to this PR which should address this; I'm going to mark this as resolved, if there are any issues let's discuss them against the newly added code.